library/Mockery/Configuration.php

```diff
@@ -127,4 +134,27 @@ public function getInternalClassMethodParamMaps()
     {
         return $this->_internalClassParamMap;
     }
+
+    public function setObjectFormatter($class, $formatterCallback)
+    {
+        $this->_objectFormatters[$class] = $formatterCallback;
+    }
+
+    public function findObjectFormatter($class)
+    {
+        $parentClass = $class;
+        for ($classes[] = $parentClass; $parentClass = get_parent_class($parentClass); $classes[] = $parentClass);
+        $classesAndInterfaces = array_merge($classes, class_implements($class));
+        foreach ($classesAndInterfaces as $type) {
+            if (isset($this->_objectFormatters[$type])) {
+                return $type;
+            }
+        }
+        return null;
```

Comment:
If we return the default here, we can remove the if/else [when calling](https://github.com/mockery/mockery/pull/766/files#diff-6326a23e84d7f7c0f0b92d031ba68bd7R594)?

```
return function ($object, $nesting) {
    return ['properties' => self::extractInstancePublicProperties($object, $nesting)];
};
```

Reply:
It is not possible because extractInstancePublicProperties is private

Reply:
Maybe pass default resolver as paramater?

Reply:
Refactored in following commit https://github.com/mockery/mockery/pull/766/commits/10cf1d7c9afeb3dd5fd22f86e46339d309880b42